### PR TITLE
Feature/rework conformance tests

### PR DESCRIPTION
**Pull request by drnlm**
This reworks the conformance tests to run the tests on surfaces with multipile depths, and also runs the conformance test suite on travis with both python 2.7 & python 3.6, to ensure we spot future regressions.

### CODE REVIEW

File: conformance/conf_tests/helpers.py [hunk 1]
```diff
@@ -21,56 +21,114 @@
...
+DEPTHS = (8, 16, 24, 32)
+
+
+class conformance_test_case(object):
+    """Tacks the appropriate attributes onto the test function"""
+
+    registry = {}
+
+    def __init__(self, test_file_name, depths=DEPTHS):
+        self.test_file_name = test_file_name
```
Comment (hodgestar) [inline conformance/conf_tests/helpers.py]: Could we get the filename from the function being wrapped by stripping `test_` from the start of it?

File: conformance/conf_tests/__init__.py [hunk 2]
```diff
@@ -38,43 +38,43 @@
...
+        test_rgb_mult,
+        test_blend_mult,
+        test_rotozoom,
+        test_scale,
+        test_flip,
+        test_flip_subsurface,
+        test_rotate,
+        test_chop,
+        test_chop_subsurface,
+]
```
Comment (hodgestar) [inline conformance/conf_tests/__init__.py]: Could we get the list from the registry somehow now?

Comment (drnlm) [inline conformance/conf_tests/__init__.py]: We could, but when working on debugging a conformance test issue, it is very useful to just restrict runs to the failing tests, so we do need to add some way of restricting the tests run if we go that route.

File: conformance/conf_tests/helpers.py (hunk 1 above)
Comment (drnlm) [inline conformance/conf_tests/helpers.py]: If we're going to base the filename off the test name, is there really any value to dropping the test_ prefix? It does result in filenames like gen_32_test_something, but all the messages will refer to the full function name, which seems like a desirable feature.

Comment (hodgestar) [inline conformance/conf_tests/helpers.py]: I'm happy to keep the full name.

File: conformance/conf_tests/__init__.py (hunk 2 above)
Comment (hodgestar) [inline conformance/conf_tests/__init__.py]: Let's leave this idea for another time then.

**Pull request merged**